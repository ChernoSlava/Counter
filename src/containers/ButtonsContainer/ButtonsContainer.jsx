import React, { useContext } from "react";
import { Button } from "@components"
import { CounterContext } from "../../contexts";

export const ButtonsContainer = () => {
    const {dispatch} = useContext(CounterContext);

    return (
        <>
            <Button title="Уменьшить" type='decrease' onClick={() => dispatch({ type: "DECREASE" })}/>
            <Button title="Сбросить" type='reset' onClick={() => dispatch({ type: "RESET" })}/>
            <Button title="Увеличить" type='increase' onClick={() => dispatch({ type: "INCREASE" })} />
        </>
        
    )
}