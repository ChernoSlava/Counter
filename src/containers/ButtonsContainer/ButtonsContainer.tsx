import React from 'react';
import { useDispatch } from 'react-redux';

import { Button } from '@components';
import { increase, decrease, reset } from '@store';

export const ButtonsContainer: React.FC = (): JSX.Element => {
  const dispatch = useDispatch();

  return (
    <>
      <Button
        title="Уменьшить"
        type="decrease"
        onClick={() => dispatch(decrease())}
      />
      <Button title="Сбросить" type="reset" onClick={() => dispatch(reset())} />
      <Button
        title="Увеличить"
        type="increase"
        onClick={() => dispatch(increase())}
      />
    </>
  );
};
