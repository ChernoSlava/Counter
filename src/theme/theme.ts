export const theme = {
  colors: {
    grey10: 'darkgrey',
    aqua10: 'aquamarine',
    red10: 'indianred',
    red20: 'red',
    black: 'black',
    green10: 'green',
  },
};

export type ThemeType = typeof theme;
