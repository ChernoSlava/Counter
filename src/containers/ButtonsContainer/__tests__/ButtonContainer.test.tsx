import { ButtonsContainer } from "../ButtonsContainer";
import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Store, AnyAction } from '@reduxjs/toolkit';

import { theme } from '../../../theme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { InitialCounter } from "@store";

const mockStore = configureStore();

const getUI = (store: Store<any, AnyAction>): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <ButtonsContainer />
      </Provider>
    </ThemeProvider>
  );
};

describe('ButtonContainer test', () => {
  it('should render', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
    }),
    ))
    expect(container).toMatchSnapshot();
  });

  it('should call decrease action', () => {
    const store = mockStore({
      ...InitialCounter,
    });

    const { getByText } = render(getUI(store))

    const button = getByText('Уменьшить');
    
    fireEvent.click(button);

    expect(store.getActions()).toMatchSnapshot();
  })
})
