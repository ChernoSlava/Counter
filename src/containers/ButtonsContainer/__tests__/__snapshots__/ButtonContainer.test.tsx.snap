// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer test should call decrease action 1`] = `
[
  {
    "payload": undefined,
    "type": "counter/decrease",
  },
]
`;

exports[`ButtonContainer test should call increase action 1`] = `
[
  {
    "payload": undefined,
    "type": "counter/increase",
  },
]
`;

exports[`ButtonContainer test should call reset action 1`] = `
[
  {
    "payload": undefined,
    "type": "counter/reset",
  },
]
`;

exports[`ButtonContainer test should render 1`] = `
.c0 {
  border: 1px solid aqua;
  border-radius: 6px;
  background-color: transparent;
  width: 140px;
  height: 47px;
  cursor: pointer;
  background-color: indianred;
}

.c0:hover {
  opacity: 0.5;
}

.c1 {
  border: 1px solid aqua;
  border-radius: 6px;
  background-color: transparent;
  width: 140px;
  height: 47px;
  cursor: pointer;
  background-color: darkgrey;
}

.c1:hover {
  opacity: 0.5;
}

.c2 {
  border: 1px solid aqua;
  border-radius: 6px;
  background-color: transparent;
  width: 140px;
  height: 47px;
  cursor: pointer;
  background-color: aquamarine;
}

.c2:hover {
  opacity: 0.5;
}

<div>
  <button
    class="c0"
  >
    Уменьшить
  </button>
  <button
    class="c1"
  >
    Сбросить
  </button>
  <button
    class="c2"
  >
    Увеличить
  </button>
</div>
`;
